client/state/plugins/installed/selectors.js

```diff
@@ -0,0 +1,164 @@
+		return plugin.sites.some( function( site ) {
+			return site.plugin && site.plugin.active;
+		} );
+	},
+	inactive: function( plugin ) {
+		return plugin.sites.some( function( site ) {
+			return site.plugin && ! site.plugin.active;
+		} );
+	},
+	updates: function( plugin ) {
+		return plugin.sites.some( function( site ) {
+			return site.plugin && site.plugin.update && site.canUpdateFiles;
+		} );
+	},
+	isEqual: function( pluginSlug, plugin ) {
+		return plugin.slug === pluginSlug;
+	}
+};
+
+const isRequesting = function( state, siteId ) {
+	// if the `isRequesting` attribute doesn't exist yet,
+	// we assume we are still launching the fetch action, so it's true
+	if ( typeof state.plugins.installed.isRequesting[ siteId ] === 'undefined' ) {
+		return true;
+	}
+	return state.plugins.installed.isRequesting[ siteId ];
+};
+
+const isRequestingForSites = function( state, sites ) {
+	// As long as any sites have isRequesting true, we consider this group requesting
+	return some( sites, ( siteId ) => isRequesting( state, siteId ) );
+};
+
+const hasRequested = function( state, siteId ) {
+	if ( typeof state.plugins.installed.hasRequested[ siteId ] === 'undefined' ) {
+		return false;
+	}
+	return state.plugins.installed.hasRequested[ siteId ];
+};
+
+const hasRequestedForSites = function( state, sites ) {
+	// All sites are requested when they all return hasRequested true
+	return every( sites, ( siteId ) => hasRequested( state, siteId ) );
+};
+
+const getPlugins = function( state, sites, pluginFilter = false ) {
+	let pluginList = {};
+	forEach( sites, ( site ) => {
+		const list = state.plugins.installed.plugins[ site.ID ] || [];
+		list.map( ( item ) => {
+			// plugin has a sites property which lists the sites this plugin is installed on
+			// each site, in turn, has a plugin property for the specific plugin's info.
+			const siteWithPlugin = Object.assign( {}, site, { plugin: item } );
+			if ( pluginList[ item.slug ] ) {
+				pluginList[ item.slug ].sites.push( siteWithPlugin );
+			} else {
+				pluginList[ item.slug ] = Object.assign( {}, item, { sites: [ siteWithPlugin ] } );
+			}
+		} );
+	} );
+	if ( !! pluginFilter && _filters[ pluginFilter ] ) {
+		pluginList = filter( pluginList, _filters[ pluginFilter ] );
+	}
+	return values( sortBy( pluginList, item => item.slug.toLowerCase() ) );
+};
+
+const getPluginsWithUpdates = function( state, sites ) {
+	const pluginList = getPlugins( state, sites );
+	if ( pluginList.length === 0 ) {
+		return [];
+	}
+	return filter( pluginList, _filters.updates );
+};
+
+const getPluginOnSite = function( state, site, pluginSlug ) {
+	const pluginList = getPlugins( state, [ site ] );
+	return find( pluginList, { slug: pluginSlug } ) || false;
+};
+
+const getSitesWithPlugin = function( state, sites, pluginSlug ) {
+	const pluginList = getPlugins( state, sites );
+	const plugin = find( pluginList, { slug: pluginSlug } );
+	if ( ! plugin ) {
+		return [];
+	}
+
+	const pluginSites = uniq( compact(
```

Comment:
What is `uniq` doing here? Seems like we shouldn't have duplicate sites within `plugin.sites` and also not sure how well it'd be able to find uniques, since they're objects.